edb/server/protocol/auth_ext/pkce.py

```diff
@@ -130,3 +142,42 @@ async def delete(db, id: str) -> None:
 
     result_json = json.loads(r.decode())
     assert len(result_json) == 1
+
+
+async def _gc(tenant: edbtenant.Tenant):
+    async with taskgroup.TaskGroup() as g:
+        for db in tenant.iter_dbs():
+            if "auth" in db.extensions:
+                g.create_task(
+                    execute.parse_execute_json(
+                        db,
+                        """
+                        delete ext::auth::PKCEChallenge filter
+                            (datetime_of_statement() - .created_at) >
+                            <duration>'10 minutes'
+                        """,
+                    ),
+                )
+
+
+async def gc(server: edbserver.BaseServer):
+    while True:
+        try:
+            tasks = [
+                tenant.create_task(_gc(tenant), interruptable=False)
+                for tenant in server.iter_tenants()
+                if tenant.accept_new_tasks
+            ]
+            if tasks:
+                done, _ = await asyncio.wait(tasks)
```

Comment:
Is there a potential of race here, when we schedule to do some work for a tenant and they are suspended before the work has started? Warning: I don't have a good grasp on how we implemented multi-tenancy.

Reply:
`accept_new_tasks` should guard us against such races, and the tenant will wait for our task to finish if started by `tenant.create_task(..., interruptable=False)`

Reply:
sounds good